core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1765,6 +1765,7 @@ private[spark] object Utils extends Logging {
    */
   def getIteratorZipWithIndex[T](iterator: Iterator[T], startIndex: Long): Iterator[(T, Long)] = {
     new Iterator[(T, Long)] {
+      require(startIndex > 0, "startIndex should be > 0.")
```

Comment:
yeah, this case inital value = -1, but fisrt generated index is 0, because there is a `index += 1` clause running first.